src/components/Select/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { PureComponent } from 'react'
+import { string, bool, array, func, object } from 'prop-types'
+import Select, { components } from 'react-select'
+
+export default class SelectInput extends PureComponent {
+  static propTypes = {
+    onChange: func.isRequired,
+    onFocus: func,
+    options: array,
+    placeholder: string,
+    isSearchable: bool,
+    defaultValue: object,
+    components: object,
+  }
+
+  static defaultProps = {
+    options: [],
+    placeholder: 'Select one',
+    isSearchable: false,
+  }
+
+  state = { isOpen: false }
+
+  handleOpenMenu = () =>
+    this.setState({ isOpen: true })
+
+  handleCloseMenu = () =>
+    this.setState({ isOpen: false })
+
+  renderDropdownIndicator = () => {
+    const { isOpen } = this.state
+    const DropdownIndicator = props =>
+      components.DropdownIndicator &&
+        <components.DropdownIndicator {...props}>
+          <Arrow rotate={isOpen} />
+        </components.DropdownIndicator>
+    return DropdownIndicator
+  }
+
+  render () {
+    const {
+      options,
+      placeholder,
+      onChange,
+      onFocus,
+      defaultValue,
+    } = this.props
+
+    return (
+      <Select
+        options={options}
+        styles={this.styles}
+        placeholder={placeholder}
+        onChange={onChange}
+        onFocus={onFocus}
+        className='mc-select'
+        classNamePrefix='mc-select__wrapper'
+        isSearchable={false}
+        components={{
+          DropdownIndicator: this.renderDropdownIndicator(),
+          ...this.props.components,
```

Comment:
I feel like, as a component library, we should probably be more decisive about how the select is rendered, and not allow the user to change how a `Select` is rendered (to this degree).

Reply:
We are using it in this scenario, in wich we need to override the default option component to add the check marks.

If we wanted to add feature with a closer component API we would need to re-implement the entire component instead of reusing this one.

<img width="469" alt="screen shot 2018-11-29 at 1 53 21 pm" src="https://user-images.githubusercontent.com/2942206/49237904-1119e880-f3de-11e8-8587-9781268bdb90.png">